install/installer/pkg/components/database/cloudsql/global.go

```diff
@@ -0,0 +1,141 @@
+// Copyright (c) 2023 Gitpod GmbH. All rights reserved.
+// Licensed under the GNU Affero General Public License (AGPL).
+// See License.AGPL.txt in the project root for license information.
+
+package cloudsql
+
+import (
+	"fmt"
+
+	"github.com/gitpod-io/gitpod/installer/pkg/cluster"
+	"github.com/gitpod-io/gitpod/installer/pkg/common"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"k8s.io/utils/pointer"
+)
+
+// Global CloudSQL is a cloud-sql-proxy which uses only one underlying database across multiple regions
+func CloudSQLGlobalObjects(ctx *common.RenderContext) ([]runtime.Object, error) {
+	if ctx.Config.Database.CloudSQLGlobal == nil {
+		return nil, nil
+	}
+
+	return common.CompositeRenderFunc(
+		cloudSQLGlobalDeployment,
+		globalCloudSQLRoleBinding,
+		common.DefaultServiceAccount(ComponentGlobal),
+		common.GenerateService(ComponentGlobal, []common.ServicePort{
+			{
+				Name:          ComponentGlobal,
+				ContainerPort: Port,
+				ServicePort:   Port,
+			},
+		}),
+	)(ctx)
+}
+
+func cloudSQLGlobalDeployment(ctx *common.RenderContext) ([]runtime.Object, error) {
```

Comment:
Sadly, these are largely duplicated with the existing cloudsql deployment. But because they are inherently different, trying to use the same definition for them would be the wrong way to go.

We also expect to be able to remove the support for cloudSQLGlobal after we complete the db-sync migration, which would make the effort to factor out common parts of the config less useful in itself